### PR TITLE
Add a new packag: sofa-c.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sofa-c/package.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class SofaC(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sofa-c/package.py]: This should be updated to use `MakefilePackage`. You can override `build_directory` to specify where the Makefile lives, override `edit` to patch the Makefile, and override `install` to copy the proper files to the installation directory.

Comment (weijianwen) [reply var/spack/repos/builtin/packages/sofa-c/package.py]: Done.

**Comment by weijianwen** [comment]
Covnerted it to a MakefilePackage. The version number `20180130` is hard-coded in `build_direcotry` setting. Is it possible to refer the version string? I tried `version`, `version[0]` and `Version`, and all failed.

File: var/spack/repos/builtin/packages/sofa-c/package.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+class SofaC(MakefilePackage):
+    "Standards of Fundamental Astronomy (SOFA) library for ANSI C."
+
+    homepage = "http://www.iausofa.org/current_C.html"
+    url      = "http://www.iausofa.org/2018_0130_C/sofa_c-20180130.tar.gz"
+
+    version('20180130', '9d6903c7690e84a788b622fba6f10146')
+
+    build_directory = join_path('20180130', 'c', 'src')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sofa-c/package.py]: Try this:
```python
@property
def build_directory(self):
    return join_path(self.version, 'c', 'src')
```

**Pull request merged**